src/app/collective-rewards/rewards/hooks/useGetLastCycleDistribution.ts

```diff
@@ -0,0 +1,36 @@
+import { Cycle } from '@/app/collective-rewards/metrics'
+import {
+  RewardDistributionFinishedEventLog,
+  useGetRewardDistributionFinishedLogs,
+} from '@/app/collective-rewards/rewards'
+
+type Log = RewardDistributionFinishedEventLog[number] & { timeStamp: number }
+export const useGetLastCycleDistribution = ({ cycleStart, cycleNext }: Cycle) => {
+  const {
+    data: rewardDistributionFinished,
+    isLoading: rewardDistributionFinishedLoading,
+    error: rewardDistributionFinishedError,
+  } = useGetRewardDistributionFinishedLogs()
+
+  const isLoading = rewardDistributionFinishedLoading
+  const error = rewardDistributionFinishedError
+
+  const [lastEvent] = rewardDistributionFinished.slice(-1) as Log[]
+
+  let fromTimestamp = cycleNext.toSeconds()
```

Comment:
```suggestion
  let fromTimestamp = cycleStart.toSeconds()
```

Reply:
Actually, I think this is on purpose. Should we had a comment to clarify it?

Reply:
Well, I wanted to discuss this with you. I'm using the next cycle because I'm sure that there are no events in the future.